lib/syskit/network_generation/merge_solver.rb

```diff
@@ -220,16 +216,49 @@ def may_merge_task_contexts?(merged_task, task)
                     return false
                 end
 
+                true
+            end
+
+            def merge_task_contexts_with_same_agent(enabled:)
+                @merge_when_identical_agents = enabled
+                r = yield
+                @merge_when_identical_agents = false
+
+                r
+            end
```

Comment:
The point of the Ruby `do_something_with_some_config { }` pattern is to
- set the configuration to some known value
- yield
- return the configuration (1) as it was and (2) in all cases (including if an exception is raised)

For (1), you would need to save the value of `@merge_when_identical_agents` and restore it afterwards to make the method really worth it. For (2), you should be guarding the yield in a begin ... ensure block:

```
def merge_task_contexts...
    old_value = @merge_when
    @merge_when = enabled
    yield
ensure
    @merge_when = old_value
end
```

I really don't like the need for this flag ... but let's see another time how we can solve it.